### ISSUE TITLE
API Endpoint produces Error 500 when ClientCertificate is Enabled

### ISSUE THREAD

**Issue by Viajaz** [ironmansoftware/powershell-universal#725]
### Steps to Reproduce

When `Authentication.ClientCertificate.Enabled` in `appsettings.json` is `true` the following API Endpoint results in error 500:
```powershell
New-PSUEndpoint -Url "/app-api/v1.0/Test" -Endpoint { @($true) | ConvertTo-Json -Depth 2 } -Authentication -Role @('Administrator') -ErrorAction "Stop"
```

### Expected behavior

```console
Normal API Endpoint behaviour
```

### Actual behavior

```console
2021-11-16 09:20:35.888 +11:00 [ERR] An unhandled exception has occurred while executing the request.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Universal.Server.Services.UniversalAuthorizationService.AuthorizedAsync(HttpContext context, String[] roles, Boolean supportChallenge) in D:\a\universal\universal\src\Universal.Server\Services\UniversalAuthorizationService.cs:line 61
   at Universal.Server.Services.ApiService.ExecuteAsync(HttpContext httpContext) in D:\a\universal\universal\src\Universal.Server\Services\ApiService.cs:line 156
   at Universal.Server.Middleware.RoutingMiddleware.Invoke(HttpContext httpContext, IPolicyEvaluator policyEvaluator) in D:\a\universal\universal\src\Universal.Server\Middleware\RoutingMiddleware.cs:line 79
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Universal.Server.Middleware.SwaggerAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next) in D:\a\universal\universal\src\Universal.Server\Middleware\SwaggerAuthMiddleware.cs:line 42
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at AspNetCoreRateLimit.RateLimitMiddleware`1.Invoke(HttpContext context) in D:\a\universal\universal\src\AspNetCoreRateLimit\Middleware\RateLimitMiddleware.cs:line 109
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2021-11-16 09:20:35.890 +11:00 [WRN] No exception handler was found, rethrowing original exception.
2021-11-16 09:20:35.890 +11:00 [ERR] Connection id "FFFFFFFFFFFFF", Request id "FFFFFFFFFFFFF:000000A1": An unhandled exception was thrown by the application.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Universal.Server.Services.UniversalAuthorizationService.AuthorizedAsync(HttpContext context, String[] roles, Boolean supportChallenge) in D:\a\universal\universal\src\Universal.Server\Services\UniversalAuthorizationService.cs:line 61
   at Universal.Server.Services.ApiService.ExecuteAsync(HttpContext httpContext) in D:\a\universal\universal\src\Universal.Server\Services\ApiService.cs:line 156
   at Universal.Server.Middleware.RoutingMiddleware.Invoke(HttpContext httpContext, IPolicyEvaluator policyEvaluator) in D:\a\universal\universal\src\Universal.Server\Middleware\RoutingMiddleware.cs:line 79
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Universal.Server.Middleware.SwaggerAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next) in D:\a\universal\universal\src\Universal.Server\Middleware\SwaggerAuthMiddleware.cs:line 42
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at AspNetCoreRateLimit.RateLimitMiddleware`1.Invoke(HttpContext context) in D:\a\universal\universal\src\AspNetCoreRateLimit\Middleware\RateLimitMiddleware.cs:line 109
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.HandleException(HttpContext context, ExceptionDispatchInfo edi)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```
```


### Environment data

PowerShell Universal 2.5.4

### Visuals

_No response_

**Comment by adamdriscoll** [comment]
We have a fix for this in 2.5.5 and I'm running a nightly build now. This also affects Windows authentication outside of IIS. 

I was able to successfully verify this with Invoke-RestMethod. 

```
PS Cert:\CurrentUser\My\> Invoke-RestMethod https://localhost:5001/authentication/role/admin
Invoke-RestMethod: Unable to read data from the transport connection: An established connection was aborted by the software in your host machine..
PS Cert:\CurrentUser\My\> Invoke-RestMethod https://localhost:5001/authentication/role/admin -CertificateThumbprint 8D2212B6EA170A33055A5BBB03FB157EF70EC93E
Authed
```

If you want to try the nightly build (available in about 30 minutes), feel free. I will continue to test and likely do a production release tomorrow of 2.5.5.

**Comment by adamdriscoll** [comment]
If you are still experiencing this issue, please feel free to reopen it.
